src/webots/gui/WbRenderingDeviceWindow.cpp

```diff
@@ -135,10 +136,22 @@ WbRenderingDeviceWindow::WbRenderingDeviceWindow(WbRenderingDevice *device) :
 }
 
 WbRenderingDeviceWindow::~WbRenderingDeviceWindow() {
+  if (!mContext)
+    return;
+
+  if (!isVisible())
+    show();  // if the window is not exposed mContext->makeCurrent() doesn't work
+
+  const bool success = mContext->makeCurrent(this);
+  assert(success);
+  if (!success)
+    return;
+
   QOpenGLFunctions_3_3_Core *f = mContext->versionFunctions<QOpenGLFunctions_3_3_Core>();
-  f->glDeleteBuffers(2, (GLuint *)&mVboId);
   f->glDeleteVertexArrays(1, &mVaoId);
-  delete mContext;
```

Comment:
I believe there is a memory leak somewhere because the `mContext` object allocated line 279 seems to be never deleted.

Reply:
No, there is no memory leak.
`mContext` is a `QObject` created with `this` as parent and Qt will automatically delete it when the parent is deleted.
So the Qt best practice (also applied in the rest of the Webots code) is to not delete it manually.

The same approach is also used in the OpenGL Window example where there is the new instruction but the delete is implicit in the destructor:
https://doc.qt.io/qt-5/qtgui-openglwindow-example.html

Reply:
Yes, you are right.